block-producer/src/txqueue/mod.rs

```diff
@@ -159,7 +159,7 @@ where
 
     /// Divides the queue in groups to be batched; those that failed are appended back on the queue
     async fn try_build_batches(&self) {
-        let txs: Vec<SharedProvenTx> = {
+        let txs: Vec<ProvenTransaction> = {
             let mut locked_ready_queue = self.ready_queue.write().await;
 
             if locked_ready_queue.is_empty() {
```

Comment:
If I remember correctly, the main reason why we used `SharedProvenTx` rather than just `ProvenTransaction` was to avoid cloning of `ProvenTransaction` objects. These objects could be pretty larger - e.g., 50KB - 100KB in size.

If I'm understanding correctly, with the current approach line 179 below would result in cloning of all transactions currently in the queue. Is the thinking that performance impact of this cloning is negligible?

Reply:
removed the copy, the `txs` are now returned with the `Error`.